### PR TITLE
Use FindWindow on ClassName instead of enumerating all windows on WindowText

### PR DESCRIPTION
**Pull request by robertcollier4**
Use FindWindow on ClassName instead of enumerating all windows on Win32. Remove unneeded Win32 code to enumerate all the windows.
SublimeText on Windows has an ASCII ClassName of "PX_WINDOW_CLASS". This can be found directly via FindWindowA function.
https://docs.microsoft.com/en-us/windows/desktop/api/winuser/nf-winuser-findwindowa

### CODE REVIEW

**Comment by gwenzek** [comment]
Thanks, this look interesting.
But assuming there are several Sublime Text windows open, which one will be returned by FindWindowA ?

**Comment by robertcollier4** [comment]
> Thanks, this look interesting.
> But assuming there are several Sublime Text windows open, which one will be returned by FindWindowA ?

https://books.google.com/books?id=WVLzCAAAQBAJ&pg=PA610&lpg=PA610&dq=findwindow+z-order
"If more than one window is found, it returns the window highest in the Z-order".
This means it will return the one most recently accessed / used. The one that was most recently at the foreground. This is the one that would have triggered an event, since all these events are based upon user interaction of the window being in the foreground which changes the title bar text. So it should be okay with multiple window instances. The FindWindowA function will always be called from the last active Sublime window, which is the one that will be found.

**Comment by bahanonu** [comment]
@robertcollier4 @gwenzek 
This works fine on my end on Windows 10, haven't run into any problems yet. Very useful. Would be great to have the option to include both file name and full path.

As a note, https://github.com/gwenzek/SublimeSetWindowTitle/tree/master __does not__ work on my end, but when I swap the files (in the same `Data\Packages\SetWindowTitle` directory) from https://github.com/robertcollier4/SublimeSetWindowTitle/tree/master then the plugin works as intended.

**Comment by bahanonu** [comment]
There is a flickering of the title (default format appears for a split second) on each file/window change before the `SetWindowTitle` title is applied, but assume that would be hard to get ride of.

**Comment by robertcollier4** [comment]
This has been tested and confirmed to work on WinXP and WIn7 by me and on Win10 by bahanonu - can this pull request be accepted?

**Comment by gwenzek** [comment]
Sorry for the delay, I wanted to test it myself before merging, but haven't found the time nor a Windows PC to do so. I'll merge this blindfolded and hope for the best 🙃

File: set_window_title.py [hunk 1]
```diff
@@ -194,17 +137,10 @@ def rename_window_linux(self, window, official_title, new_title, debug=False):
...
-        if w.title.endswith(official_title):
-          w.title = new_title
-          # TODO: Understand why caching the windows handle cause crashes.
-          # self.window_handle_cache[window.id()] = w
-    else:
-      w.title = new_title
-
+  def rename_window_windows(self, new_title):
+    # PX_WINDOW_CLASS is the ClassName of SublimeText, can be seen via a tool such as Nirsoft WinLister
+    hwndSublime = ctypes.windll.user32.FindWindowA(b'PX_WINDOW_CLASS', None)
```
Comment (gwenzek) [inline set_window_title.py:142]: I think this is probably what is causing #15.
If FindWindowA returns NULL, we will call SetWindowText(null, new_title) and probably crash.

Comment (robertcollier4) [reply set_window_title.py:142]: we can add a if(hwndSublime): below this line in this case instance

**Pull request merged**